From b7effcfc6e3f8179c8c02b05c329e1b4e30096c8 Mon Sep 17 00:00:00 2001
From: Tang Yizhou <tangyizhou@huawei.com>
Date: Fri, 3 Dec 2021 18:15:49 +0800
Subject: [PATCH] signal: Export tracepoint symbol signal_generate
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: b7effcfc6e3f8179c8c02b05c329e1b4e30096c8


hulk inclusion
category: feature
bugzilla: 37631
CVE: NA

-------------------------------------------------

To support signal monitoring, we need to export the defined tracepoint
signal_generate so that it can be used in kernel modules.

Signed-off-by: Tang Yizhou <tangyizhou@huawei.com>
Reviewed-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Reviewed-by: Weilong Chen <chenweilong@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/signal.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/signal.c b/kernel/signal.c
index 30e1b37a73e1..a6434ee9cdbb 100644
--- a/kernel/signal.c
+++ b/kernel/signal.c
@@ -56,6 +56,8 @@
 #include <asm/siginfo.h>
 #include <asm/cacheflush.h>
 
+EXPORT_TRACEPOINT_SYMBOL(signal_generate);
+
 /*
  * SLAB caches for signal bits.
  */
-- 
2.26.2

